src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -625,6 +719,23 @@ private class FastqDemultiplexer(val sampleInfos: Seq[SampleInfo],
     case n => throw new IllegalArgumentException(s"$n template reads defined. Can't process > 2 template reads.")
   }
 
+  /** Gets the two items [[T]] with the largest associated values ([[Int]]). */
+  private def getTopTwoByMaximum[T](elems: TraversableOnce[(T, Int)]): (Option[(T, Int)], Option[(T, Int)]) = {
```

Comment:
I found this sped things up 10-20%!

Reply:
Nice!  I think this is a perfect example of where a tiny case class e.g. `case class IndexAndMatches(index: String, matches: Int)` would make this signature and a bunch of the matches below _way_ clearer.